context/TxContext.tsx

```diff
@@ -24,15 +30,22 @@ export const TxProvider = ({ children }: PropsWithChildren) => {
 
   const { openModal } = useModal();
 
-  const submitTx = async (args: WriteContractParameters): Promise<void> => {
+  const submitCustomTx = async ({
+    writeTxFunction,
+    args,
+  }: any): Promise<Address | void> => submitWriteTx(() => writeTxFunction(args));
```

Comment:
```suggestion
  const submitCustomTx = async (tx: Function): Promise<void> => submitWriteTx(() => tx());
```

Reply:
Here the function name should be `writeTxFn`instead of only `tx`